flink-connectors/flink-connector-elasticsearch-base/src/test/java/org/apache/flink/connector/elasticsearch/sink/ElasticsearchWriterITCase.java

```diff
@@ -113,7 +113,8 @@ void testWriteOnBulkFlush() throws Exception {
             writer.write(Tuple2.of(4, buildMessage(4)), null);
 
             // Ignore flush on checkpoint
-            writer.prepareCommit(false);
+            // TODO: clarify if it needs to be emulated with flush()
+            // writer.prepareCommit(false);
```

Comment:
Oh yeah, that makes sense, I missed that bit (`flushOnCheckpoint = false`). The existing test was not really checking the `flushOnCheckpoint` in isolation, because the call was made  with  prepareCommit(**false**), i.e. disabled by two variables at the same time.